File: trunk/unityClientSample/Assets/src/protocol/BaseInfo.cs
using ProtoBuf;
using System.Collections.Generic;
using System;

namespace protocol
{
	[ProtoContract]
	public class BaseInfo
	{
        /// <summary>
        ///  基地id
        /// </summary>
		[ProtoMember(1, IsRequired = false)]
		public string id;
        /// <summary>
        ///  名称
        /// </summary>
		[ProtoMember(2, IsRequired = false)]
		public string name;
        /// <summary>
        ///  血量
        /// </summary>
		[ProtoMember(3, IsRequired = false)]
		public int blood;
        /// <summary>
        ///  消耗
        /// </summary>
		[ProtoMember(4, IsRequired = false)]
		public int cost;
        /// <summary>
        ///  科技等级
        /// </summary>
		[ProtoMember(5, IsRequired = false)]
		public int science;
        /// <summary>
        ///  技能
        /// </summary>
		[ProtoMember(6, IsRequired = false)]
		public string skill;
        /// <summary>
        ///  美术资源
        /// </summary>
		[ProtoMember(7, IsRequired = false)]
		public string art;
        /// <summary>
        ///  描述
        /// </summary>
		[ProtoMember(8, IsRequired = false)]
		public string desc;

	}
}

File: trunk/server/src/main/java/org/ngame/protocol/domain/BaseInfo.java
/**
 * 基地
 */
package org.ngame.protocol.domain;

import lombok.Data;





import com.baidu.bjf.remoting.protobuf.annotation.Protobuf;

@Data
public class BaseInfo
{
  @Protobuf(description = "基地id")
  private String id;// id
  @Protobuf(description = "名称")
  private String name;// 名称
  @Protobuf(description = "血量")
  private int blood;
  @Protobuf(description = "消耗")
  private int cost;
  @Protobuf(description = "科技等级")
  private int science;
  @Protobuf(description = "技能")
  private String skill;
  @Protobuf(description = "美术资源")
  private String art;
  @Protobuf(description = "描述")
  private String desc;
  public BaseInfo()
  {
  }

  public BaseInfo(String id)
  {
    this.id = id;
  }
  
  public void addScience(int science){
	  this.science+=science;
  }
  
  public boolean subBlood(int hurt){
	  blood=blood-hurt <0 ? 0 : blood-hurt;
	  return blood == 0;
  }
  
}
